packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -60,6 +60,22 @@ abstract class FlutterCommand extends Command {
     _usesPubOption = true;
   }
 
+  void addBuildModeFlags() {
+    argParser.addFlag('debug',
```

Comment:
`--build-mode=debug` would be clearer but more to type. I'll add some checks in the `getBuildMode()` call below to handle mutually exclusive flags.